### PR TITLE
'unicode' does not have the buffer interface

### PR DESCRIPTION
**Pull request by krischer**
1.0.3.post0+946.g1ab414164e.obspy.master
macos 10.12.4

I want to convert a dataless to RESP.

```code
rdseed -f dataless.G.SSB.seed -R
````
works without problem. But with Obspy I have an error:

```code
from obspy.io.xseed import Parser
p=Parser('http://geoscope.ipgp.fr/metadata/G/dataless.G.SSB.seed')
p.get_resp()
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-8-c939779e0345> in <module>()
----> 1 p.get_resp()

/Users/bonaime/git/obspy/obspy/io/xseed/parser.pyc in get_resp(self)
    387                     if _len != 0:
    388                         # Send the blockettes to the parser and append to list.
--> 389                         self._get_resp_string(resp, blockettes, cur_station)
    390                         resp_list.append([filename, resp])
    391                     # Create the file name.

/Users/bonaime/git/obspy/obspy/io/xseed/parser.pyc in _get_resp_string(self, resp, blockettes, station)
    846             try:
    847                 resp.write(blkt.get_resp(
--> 848                     station, channel_info['Channel'], self.abbreviations))
    849             except AttributeError:
    850                 msg = 'RESP output for blockette %s not implemented yet.'

TypeError: 'unicode' does not have the buffer interface
```

### CODE REVIEW

**Comment by krischer** [comment]
This is now a PR fixing it.

There were actually quite some issues on Python 3 due a test case that did not actually do anything. This is now fixed, additionally a new test case explicitly testing all `get_resp()` methods and their return types has been added. A unit test for blockette 62 has also been added.

**Comment by krischer** [comment]
IMHO good to go assuming CI passes.

**Comment by krischer** [comment]
Done from my side. @bonaime @dsentinel Can one of you please review it? Thanks :)

**Comment by bonaime** [comment]
Good to merge !

File: obspy/io/xseed/tests/test_blockettes.py [hunk 1]
```diff
@@ -40,7 +44,7 @@ def parse_file(self, blkt_file):
         # Create a new empty list to store all information of the test in it.
         test_examples = []
         # Now read the corresponding file and parse it.
-        file = open(blkt_file, 'rb')
+        file = io.open(blkt_file, 'rt')
```
Comment (megies) [inline obspy/io/xseed/tests/test_blockettes.py]: Might be good to use `with` contruct..

Comment (megies) [inline obspy/io/xseed/tests/test_blockettes.py]: Ah, merged already.. nevermind..

Comment (krischer) [inline obspy/io/xseed/tests/test_blockettes.py]: Done in bee249453fb793115728e5b17a2278e92e94fa59

**Pull request merged**